lib/LaTeXML/Package.pm

```diff
@@ -209,23 +209,33 @@ sub parseParameters {
   $p =~ s/^\s+//; $p =~ s/\s+$//;
   while ($p) {
     # Handle possibly nested cases, such as {Number}
+    # That can be explicitly defined as {Number},
+    # OR will be handled as readArg, then the content reparsed as Number
     if ($p =~ s/^(\{([^\}]*)\})\s*//) {
       my ($spec, $inner_spec) = ($1, $2);
-      my $inner = ($inner_spec ? parseParameters($inner_spec, $for) : undef);
-      # If single inner spec is optional, make whole thing optional
-      my $opt = $inner && (scalar(@$inner) == 1) && $$inner[0]{optional};
-      push(@params, LaTeXML::Core::Parameter->new('Plain', $spec, extra => [$inner],
-          optional => $opt)); }
+      if(LookupMapping('PARAMETER_TYPES', $spec)) { # If specially defined with braces?
+        push(@params, LaTeXML::Core::Parameter->new($spec, $spec)); }
+      else {
+        my $inner = ($inner_spec ? parseParameters($inner_spec, $for) : undef);
+        # If single inner spec is optional, make whole thing optional
+        my $opt = $inner && (scalar(@$inner) == 1) && $$inner[0]{optional};
+        push(@params, LaTeXML::Core::Parameter->new('Plain', $spec, extra => [$inner],
+          optional => $opt)); } }
     elsif ($p =~ s/^(\[([^\]]*)\])\s*//) {    # Ditto for Optional
+      # Optional also can be defined explicitly as [Type]
+      # OR read using readOptional, then content (if any) reparsed as Type.
       my ($spec, $inner_spec) = ($1, $2);
-      if ($inner_spec =~ /^Default:(.*)$/) {
-        push(@params, LaTeXML::Core::Parameter->new('Optional', $spec,
-            extra => [TokenizeInternal($1), undef])); }
-      elsif ($inner_spec) {
-        push(@params, LaTeXML::Core::Parameter->new('Optional', $spec,
-            extra => [undef, parseParameters($inner_spec, $for)])); }
+      if(LookupMapping('PARAMETER_TYPES', $spec)) { # If specially defined with []?
+        push(@params, LaTeXML::Core::Parameter->new($spec, $spec, optional => 1)); }
```

Comment:
I am now starting to get confused about having two ways of specifying these. I recall we also had a prefix keyword `Optional`, as with `OptionalNumber`, as well as `[Number]`.

Shouldn't we be normalizing towards one or the other? Is there a risk that a lookup for `OptionalNumber` doesn't find a special mapping for `[Number]` or the other way around?

Reply:
Good point, but there's are Lots of the latter